### PR TITLE
feat: add support for ocp 4.15

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Description

Issue : https://github.ibm.com/GoldenEye/issues/issues/8745

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: patterns/roks/variables.tf [hunk 1]
```diff
@@ -192,6 +193,12 @@ variable "manage_all_cluster_addons" {
   description = "Instructs Terraform to manage all cluster addons, even if addons were installed outside of the module. If set to 'true' this module will destroy any addons that were installed by other sources."
 }
 
+variable "disable_outbound_traffic_protection" {
+  type        = bool
+  description = "Whether to allow public outbound access from the cluster workers. This is only applicable for `ocp_version` 4.15"
```
Comment (ocofaigh) [inline patterns/roks/variables.tf:198]: hmm, `ocp_version` is not the variable name for SLZ. Maybe use English here and say "Red Hat OpenShift 4.15"

File: ibm_catalog.json [hunk 2]
```diff
@@ -1121,6 +1121,10 @@
                 {
                   "displayname": "4.14_openshift",
                   "value": "4.14_openshift"
+                },
+                {
+                  "displayname": "4.15_openshift",
+                  "value": "4.15_openshift"
```
Comment (ocofaigh) [inline ibm_catalog.json:1127]: I think update the `default_value` to 4.15 so the tests will run it. We won't release it to prod for another month, so if there are issues, plenty of time to debug

File: patterns/roks/module/variables.tf [hunk 3]
```diff
@@ -184,6 +184,12 @@ variable "manage_all_cluster_addons" {
   description = "Instructs Terraform to manage all cluster addons, even if addons were installed outside of the module. If set to 'true' this module will destroy any addons that were installed by other sources."
 }
 
+variable "disable_outbound_traffic_protection" {
+  type        = bool
+  description = "Whether to allow public outbound access from the cluster workers. This is only applicable for `ocp_version` 4.15"
```
Comment (ocofaigh) [inline patterns/roks/module/variables.tf:189]: hmm, `ocp_version` is not the variable name for SLZ. Maybe use English here and say "Red Hat OpenShift 4.15"

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Review by ocofaigh — APPROVED** [review]

**Comment by Aashiq-J** [comment]
```
2024-05-14T16:52:14.6911615Z     destroy.go:11: 
2024-05-14T16:52:14.6912497Z         	Error Trace:	/go/pkg/mod/github.com/gruntwork-io/terratest@v0.46.13/modules/terraform/destroy.go:11
2024-05-14T16:52:14.6913766Z         	            				/__w/terraform-ibm-landing-zone/terraform-ibm-landing-zone/tests/pr_test.go:509
2024-05-14T16:52:14.6914606Z         	Error:      	Received unexpected error:
2024-05-14T16:52:14.6915506Z         	            	FatalError{Underlying: error while running command: exit status 1; [31m╷[0m[0m
2024-05-14T16:52:14.6917244Z         	            	[31m│[0m [0m[1m[31mError: [0m[0m[1mtimeout while waiting for state to become 'done, failed' (last state: 'deleting', timeout: 10m0s)[0m
2024-05-14T16:52:14.6918332Z         	            	[31m│[0m [0m
2024-05-14T16:52:14.6918778Z         	            	[31m│[0m [0m[0m
2024-05-14T16:52:14.6919153Z         	            	[31m╵[0m[0m}
2024-05-14T16:52:14.6919577Z         	Test:       	TestRunVsiExtention
2024-05-14T16:52:14.6920011Z --- FAIL: TestRunVsiExtention (1243.28s)
```
Error not related to this PR change.
Retrying..

**Comment by ocofaigh** [comment]
@Aashiq-J The tests are going to fail due to known issue with destroy - since its unrelated to your changes, the since the OCP tests passed, I'm going to force merge this PR

**Pull request merged**